Revised topics about nullable types

pkulikov opened the pull request:
Revised two topics about nullable types (probably, the best way to review is side by side):
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types

Moved snippets to the samples repo: supported by dotnet/samples#205

Fixes #6630

pkulikov committed:
Updated link to the code snippet

pkulikov committed:
Updated using-nullable-types.md

BillWagner commented:
closing and reopening to force a new build

BillWagner reviewed (changes requested):
I really like the work you've done here @pkulikov 
I had a number of small suggestions, and then it's ready.

Thanks again for all the work.

pkulikov commented on docs/csharp/programming-guide/nullable-types/using-nullable-types.md:
```diff
@@ -1,97 +1,88 @@
--   `HasValue`  
+## Explicit conversions
+
+A nullable type can be cast to a regular type, either explicitly with a cast, or by using the `Value` property. For example:  
```
@BillWagner then the **Explicit conversions** section starts to overlap with the **Assignment to non-nullable type** section, which is further in the topic. I suggest to merge those two then and name it **Assignment to non-nullable type**.

Furthermore, the **Implicit conversions** section can be removed, as it almost duplicates the first section: **Declaration and assignment**.

That way, the whole topic would become less fragmented containing only the following sections:
 - Declaration and assignment of nullable type
 - Assignment to non-nullable type
 - Operators
 - The bool? type

What do you think?

pkulikov committed:
Addressed feedback in index.md

pkulikov committed:
Made using-nullable-types.md more compact

pkulikov commented:
@BillWagner thank you for the review. I've addressed your feedback for the *index.md* file. As for the *using-nullable-types.md* file, I've given it one more look and updated the division into sections. Please review it again.

The latest commit relies on the dotnet/samples#209

pkulikov committed:
Minor edits

pkulikov committed:
Mention pattern matching

pkulikov commented:
@BillWagner I've updated the "Using nullable types" topic with pattern matching case.

pkulikov committed:
Mention pattern matching in the overview

BillWagner commented:
closing and reopening to start a new build, to include the new sample)

BillWagner commented:
Thanks for making the updates @pkulikov I've reviewed the latest changes, and I'll :shipit: now.

Thanks again.
